### ISSUE TITLE
CSS cache not cleared properly in developer mode

### ISSUE THREAD

**Issue by adamlavery** [magento/magento2#32470]
### Summary (*)
When in developer mode, using the option in Cache Management to flush JS/CSS cache and flush Static Files Cache should clear all caches and enable **all** changes to be viewed on next refresh.

### Examples (*)
Leave LESS compilation as default (server side)
Make a change to a module .less file
Flush the JS/CSS, static files and full page caches from the dashboard.

**What should happen:** 
Refresh the page and the style rule changes should show up.

**What does happen:**
The changes do not show up.

### Proposed solution
Instead of including a note in the documentation that the the var/view_preprocessed folder must also be manually emptied, empty the folder when flushing the JS/CSS cache. Then the CSS cache is fully-flushed instead of half-flushed.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users with no workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by mrtuvn** [comment]
For dev enviroments 
i should use grunt workflow to clean up or watch change instead manually way (clear folder by hand or go to admin config for flush cache). Rarelly use button "Flush Static Files Cache". This may use in production cases

Less process in magento little difference than you expected. Especially in developer enviroment. Example your file less in app/code/YourModule. When you change without do some deploy/compile less magento still get files in pub/static/ folder. The command that you use for static deploy is run some compile less and symlink it to var/view_preprocessed folder and link to pub/static/ in eventually. You can try modify css in folder pub/static/ then reload page. Your change will update immediately

**Comment by adamlavery** [comment]
Point is there should be no need for any of this messing about to get changes published. No workarounds would be required if clearing the static file cache actually cleared the static files cache.

**Comment by adamlavery** [comment]
ping

**Comment by engcom-Echo** [comment]
Hi @adamlavery,

We were not able to re-produce with the steps given . We followed the steps in our case the changes that we did to our module_.less file was reflected after flushing cache, static page and full page without requiring to manually delete var/pre_processed files.
Requesting you to kindly test the same on latest branch i.e. 2.4-develop and let us with detailed input if you were able to re-produce the same on that branch.

Thanks
